client/my-sites/upgrades/checkout/secure-payment-form.jsx

```diff
@@ -186,46 +134,113 @@ var SecurePaymentForm = React.createClass( {
 		}
 	},
 
-	selectPaymentBox: function( paymentBox ) {
+	selectPaymentBox( paymentBox ) {
 		this.setState( {
 			userSelectedPaymentBox: paymentBox,
 			visiblePaymentBox: paymentBox
 		} );
-	}
-} );
+	},
+
+	renderCreditsPayentBox() {
+		return (
+			<CreditsPaymentBox
+				cart={ this.props.cart }
+				onSubmit={ this.handlePaymentBoxSubmit }
+				transactionStep={ this.props.transaction.step } />
+		);
+	},
 
-SecurePaymentForm.Placeholder = React.createClass( {
-	displayName: 'SecurePaymentForm.Placeholder',
+	renderFreeTrialConfirmationBox() {
+		return (
+			<FreeTrialConfirmationBox
+				cart={ this.props.cart }
+				onSubmit={ this.handlePaymentBoxSubmit }
+				transactionStep={ this.props.transaction.step } />
+		);
+	},
 
-	render: function() {
+	renderFreeCartPaymentBox() {
 		return (
-			<PaymentBox
-				classSet="selected is-empty"
-				contentClassSet="selected is-empty" >
-				<div className="payment-box-section">
-
-					<div className="placeholder-row placeholder"/>
-					<div className="placeholder-row placeholder"/>
-					<div className="placeholder-col-narrow placeholder-inline-pad">
-						<div className="placeholder" />
-					</div>
-					<div className="placeholder-col-narrow placeholder-inline-pad-only-wide">
-						<div className="placeholder" />
-					</div>
-					<div className="placeholder-col-wide">
-						<div className="placeholder" />
-					</div>
-					<div className="placeholder-row placeholder"/>
-				</div>
-				<div className="payment-box-hr" />
-				<div className="placeholder-button-container">
-					<div className="placeholder-col-narrow">
-						<div className="placeholder placeholder-button"></div>
-					</div>
-				</div>
-			</PaymentBox>
+			<FreeCartPaymentBox
+				cart={ this.props.cart }
+				onSubmit={ this.handlePaymentBoxSubmit }
+				products={ this.props.products }
+				selectedSite={ this.props.selectedSite }
+				transactionStep={ this.props.transaction.step } />
+		);
+	},
+
+	renderCreditCardPaymentBox() {
+		return (
+			<CreditCardPaymentBox
+				cards={ this.props.cards }
+				transaction={ this.props.transaction }
+				cart={ this.props.cart }
+				countriesList={ countriesListForPayments }
+				initialCard={ this.getInitialCard() }
+				selectedSite={ this.props.selectedSite }
+				onToggle={ this.selectPaymentBox }
+				onSubmit={ this.handlePaymentBoxSubmit }
+				transactionStep={ this.props.transaction.step } />
+		);
+	},
+
+	renderPayPalPaymentBox() {
+		return (
+			<PayPalPaymentBox
+				cart={ this.props.cart }
+				transaction={ this.props.transaction }
+				countriesList={ countriesListForPayments }
+				selectedSite={ this.props.selectedSite }
+				onToggle={ this.selectPaymentBox }
+				redirectTo={ this.props.redirectTo } />
+		);
+	},
+
+	getPaymentBox() {
```

Comment:
It's much more intuitive after this refactoring 👍 
This CSS visibility approach was hard to understand.